docs/getting-started.rst

```diff
@@ -0,0 +1,249 @@
+  placing the new class in. The ``class`` is whatever you've called the class.
+* Design the class following these guidelines:
+    * The constructor should always accept, before any other arguments, the following positional options:
+        * ``name``: The internal name of the resource as Pulumi tracks it.
+        * ``project``: The ThunderbirdPulumiProject these resources belong to.
+    * The constructor should always accept the following keyword arguments:
+        * ``opts``: A ``pulumi.ResourceOptions`` object which will get merged into the default set of arguments managed
+          by the project.
+    * The constructor should explicitly define only those arguments that you intend to have default values which differ
+      from the default values the provider will set, or which imply larger patterns (such as ``build_jumphost`` implying
+      other resources, like a security group and its rules, not just an EC2 instance).
+    * The constructor may accept a final ``**kwargs`` argument with arbitrary meaning. Because the nature of a component
+      resource is to compile many other resources into one class, it is not implicitly clear what "everything else"
+      should apply to. If this is implemented, its function should be clearly documented in the class.
+    * The class should extend ``tb_pulumi.ThunderbirdComponentResource``.
+    * The class should call its superconstructor in the following way:
+        * ``super().__init__(typestring, name, project, opts=opts)``
+    * Any resources you create should always be assigned a key in ``self.resources``.
+    * Any resources you create must have the ``parent=self`` ``pulumi.ResourceOption`` set.
+    * At the end of the ``__init__`` function, you must call ``self.finish()``
+
+
+Troubleshooting
+---------------
+
+
+Pythonic problems
+^^^^^^^^^^^^^^^^^
+
+This Pulumi code is developed against Python 3.12 or later. If this is not your default version, you'll need to manage
+your own virtual environment.
+
+Check your default version:
+::
+
+  $ python -V
+  Python 3.12.6
+
+If you need a newer Python, `download and install it <https://www.python.org/downloads/>`_. Then you'll have to set up
+the virtual environment yourself with something like this:
+::
+
+  virtualenv -p /path/to/python3.12 venv
+  ./venv/bin/pip install -r requirements.txt
+
+After this, ``pulumi`` commands should work. If 3.12 is your default version of Python, Pulumi should set up its own
+virtualenv, and you should not have to do this.
+
+
+Shells other than Bash
+^^^^^^^^^^^^^^^^^^^^^^
+
+Setup instructions in these docs are designed for use with the Bourne Again SHell (Bash). Pulumi also seems to make some
+assumptions like this when it installs itself. Pulumi will install itself into a hidden folder in your home directory:
+``~/.pulumi/bin``. You may need to add this to your ``$PATH`` to avoid having to make the explicit reference with every
+``pulumi`` command.
```

Comment:
Bleh more home directory pollution. Nothing we can do here though 😅

Reply:
There is probably an advanced Pulumi option or env-var you can set to have it use some other directory.